### PR TITLE
Add LaTeX repr for distributions

### PR DESCRIPTION
**Pull request by fonnesbeck**
This PR adds LaTeX output when distributions are shown in Jupyter notebooks. The initial commit has this implemented for all continuous variables. Discrete, MV to come.

### CODE REVIEW

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -283,6 +307,12 @@ def logp(self, value):
                      value >= 0,
                      tau > 0, sd > 0)
 
+    def _repr_latex_(self, name=None, dist=None):
+        if dist is None:
+            dist = self
+        sd = dist.sd
+        return r'$%s \sim \text{HalfNormal}(\mathit{sd}=%s)$' % (name,
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: how about new-style `r'{}'.format(...)`? https://pyformat.info/

**Comment by fonnesbeck** [comment]
Odd failures related to multiprocessing in 2.7.

File: pymc3/distributions/continuous.py (hunk 1 above)
Comment (fonnesbeck) [reply pymc3/distributions/continuous.py]: The new style does not like strings with curly braces in them. This was easier.

Comment (fonnesbeck) [reply pymc3/distributions/continuous.py]: OK, figured it out.

**Comment by ghost** [comment]
The failures occur because of usage `functools.partial` with bound method as an argument.

Multiprocessing tries to pickle resulting function objects, but fails because bound methods are not pickleable in Python 2, although they are pickleable in Python 3. For example, this code works with Python 3, but not Python 2:

```python
import pickle
class Foo:
    def bar(self):
        return 3
pickle.dumps(Foo().bar)
```

I tried to replace invocations of `functools.partial` by direct implementations of `_repr_latex_` method in [my fork](https://github.com/a-rodin/pymc3/tree/latex_dists) and the tests seem to work there.

**Comment by twiecki** [comment]
Thanks @a-rodin. The best approach is to do a PR onto this PR (specifically, the `latex_dists` branch) with your fixes.

File: pymc3/distributions/timeseries.py [hunk 2]
```diff
@@ -70,29 +79,38 @@ def __init__(self, tau=None, init=continuous.Flat.dist(), sd=None, mu=0.,
...
+        return r'${} \sim \text{{GaussianRandomWalk}}(\mathit{{mu}}={}, \mathit{{sd}}={})$'.format(name,
+                                                get_variable_name(mu),
+                                                get_variable_name(sd))
 
 
 class GARCH11(distribution.Continuous):
     """
     GARCH(1,1) with Normal innovations. The model is specified by
-
+
```
Comment (twiecki) [inline pymc3/distributions/timeseries.py]: I think these should not be here, not quite sure.

**Comment by twiecki** [comment]
Looks like the whitespaces are still there.

**Comment by fonnesbeck** [comment]
It appears that My Favorite Text Editor sucks at whitespace removal, even inserting it in some places it did not exist.

**Comment by fonnesbeck** [comment]
Should be good to go now.

**Comment by ColCarroll** [comment]
would this be reasonable to propagate to models and maybe even the traces?  It looks like it is more work than `'\n'.join([repr(v) for v in self.vars])`.

**Comment by fonnesbeck** [comment]
I think having a `_repr_latex_` for a model would be a great idea. Would be a nice way to get an overview of the model, particularly if it is built up over a number of cells.

**Comment by ferrine** [comment]
vote for renaming `_repr_latex_` to `__latex__`. The reason is that single leading underscore denotes *private* attributes and methods. But `_latex_repr_` is not private at all. As for `__latex__` it is our magic.

**Comment by fonnesbeck** [comment]
We aren’t at liberty to change it because they are hooks used by IPython. If you change the name they won’t work.

**Pull request merged**